otherlibs/stdune/src/string.ml

```diff
@@ -278,7 +278,17 @@ let need_quoting s =
 ;;
 
 let quote_for_shell s = if need_quoting s then Stdlib.Filename.quote s else s
-let quote_list_for_shell l = List.map l ~f:quote_for_shell |> concat ~sep:" "
+
+let quote_list_for_shell = function
+  | [] -> ""
+  | prog :: args ->
+    let prog =
+      if Sys.win32 && contains prog '/'
+      then concat ~sep:"\\" (split_on_char ~sep:'/' prog)
+      else prog
+    in
+    prog :: List.map ~f:quote_for_shell args |> concat ~sep:" "
```

Comment:
`prog` is no longer quoted_for_shell. Is it expected?

Reply:
Thanks!